app/bot.py

```diff
@@ -1,77 +1,138 @@
+
+    SELECTED_CHANNEL = client.guilds[selected_index].channels[selected_channel_index]
+    #await client.guilds[selected_index].channels[selected_channel_index].send("Discord Bot Online.")
+    print(SELECTED_CHANNEL)
+       
+
+@client.event
+async def on_message(message):
+    print(
+        "{}'s Message: {} Channel: {} ".format(
+            message.author,
+            message.content,
+            message.channel,
+        )
+    )
+
+    if message.author == client.user:
+        return 
+
+    if message.content.startswith('$replace'):
+        await message.channel.send('Conducting an act of Replacement.')
+    
+    # messages are for test purpose only. insert a more 'formal' message for actual use.
+    randomMessage = [
+        "busy pa si boss Carl, nagkakape. reply na lang siya mamaya.", 
+        "naghuhugas pa ng pinggan si Carl. jk babalik 'yon maya-maya.",
+        "tulog pa ata, gisingin ko wait. chour rreply nalang siya mamaya"
+    ]
+
+    randomGreeting = [
+        "Yoww",
+        "'Sup",
+        "Hii",
+        "Hallu"
+    ]
+    # if the author is mentioned, and send back a message. 
+    # replace @!723498127944056845 with the ID of the user that is needed to be checked for mentions.
+    userid=int(USERID)
+    # `@!` is needed to be added to the user ID to check for mentions.
+    mentionid=f'@!{userid}'
+    if mentionid in message.content:
+        # option for send in a custom message
+    #   if BOT_MODE == False: 
+    #       response = input('Type in your reply: ')
+    #       await message.channel.send(f"{message.author.mention}, {response}") 
+    #   else:
+            
+            # get the member
+            guild = message.author.guild
+            member = discord.Guild.get_member(self=guild, user_id=userid)
+
+            # checks if the user's status is online. 
+            if member.status is not discord.Status.online: 
+                response = random.choice(randomMessage)
+                greet = random.choice(randomGreeting)
+                msgStr = (f"{greet} {message.author.mention}!, {response}")
+                fromDescription = (f"A reply to {member.nick}'s message!")
+                embed = discord.Embed(
+                    title = None,
+                    description = None,
+                    colour = discord.Color.blue()
+                )
+                embed.add_field(name=fromDescription,value=message.content,inline=False)
+                await message.channel.send(embed =embed,content=msgStr)
+                # previous method, no embedding of message. only mentions the author.
+                # await message.channel.send(f"{greet} {message.author.mention}!, {response}") 
+    
+    
+    #responds to mentions - from BoredPaper's PR
+    #for x in message.mentions:
+    #    if (x == client.user):
+    #        await message.channel.send(f"{message.author.mention} hello there")
+        
+
+# @client.event
+# async def on_member_join(member):

```

Comment:
why comment this out?

Reply:
are you referring to lines 125-128? it was a mistake, i wasn't able to remove the comment tags before uploading.

Reply:
uncomment it